google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ServerToIDEFileResolver.java

```diff
@@ -75,11 +76,15 @@ public VirtualFile getFileFromPath(@NotNull Project project, @NotNull String pat
     VirtualFile file = FILE_SYSTEM.findFileByPath(project.getBasePath() + "/" + path);
     // Try class name with package and class file name.
     if (file == null) {
-      PsiClass[] matchingClasses = JavaPsiFacade.getInstance(project)
-          .findPackage(getPackageFromPath(path))
-          .findClassByShortName(getClassNameFromPath(path), GlobalSearchScope.allScope(project));
-      if (matchingClasses.length > 0) {
-        file = matchingClasses[0].getContainingFile().getVirtualFile();
+      PsiPackage psiPackage = JavaPsiFacade.getInstance(project)
+          .findPackage(getPackageFromPath(path));
+      // If a class isn't in a project's classpath, psiPackage will be null.
+      if (psiPackage != null) {
+        PsiClass[] matchingClasses = psiPackage.findClassByShortName(
```

Comment:
It gets the wrong class if two classes with the same package and class name exist in the same project. I think this is very unusual and can only happen if there is more than one library, with two or more classes having the same package and name.

If this were to happen, we wouldn't have any way of deciding what the right class would be and would have to ask the user for. Then there would also be a chance the user wouldn't know which class that would be...